### ISSUE TITLE
Multi-gpus call failed by  tf.distribute.MirroredStrategy()

### ISSUE THREAD

**Issue by PureHing** [tensorflow/tensorflow#44030]
**System information**
- Linux Ubuntu 18.04
- TensorFlow installed from (source or binary):pip install tf-nightly-gpu
- TensorFlow version (use command below):2.4.0-dev20201014
- Python version:3.7.8
- CUDA/cuDNN version:CUDA11.1+CUDNN8.0.4.30
- Multi GPUs: RTX3080*2

**Describe the current behavior**
Code
```sh
import tensorflow as tf
import tensorflow_datasets as tfds

num_epochs = 5
batch_size_per_replica = 64
learning_rate = 0.001
strategy = tf.distribute.MirroredStrategy()
print('Number of devices: %d' % strategy.num_replicas_in_sync)  
batch_size = batch_size_per_replica * strategy.num_replicas_in_sync


def resize(image, label):
    image = tf.image.resize(image, [224, 224]) / 255.0
    return image, label

dataset = tfds.load("cats_vs_dogs", split=tfds.Split.TRAIN, as_supervised=True)
dataset = dataset.map(resize).shuffle(1024).batch(batch_size)

with strategy.scope():
    model = tf.keras.applications.MobileNetV2(weights=None, classes=2)
    model.compile(
        optimizer=tf.keras.optimizers.Adam(learning_rate=learning_rate),
        loss=tf.keras.losses.sparse_categorical_crossentropy,
        metrics=[tf.keras.metrics.sparse_categorical_accuracy]
    )

model.fit(dataset, epochs=num_epochs)
```
<details>
<summary>The error log is as follows</summary>
<pre><code>
Epoch 1/5
INFO:tensorflow:batch_all_reduce: 158 all-reduces with algorithm = nccl, num_packs = 1
INFO:tensorflow:batch_all_reduce: 158 all-reduces with algorithm = nccl, num_packs = 1
INFO:tensorflow:batch_all_reduce: 158 all-reduces with algorithm = nccl, num_packs = 1
INFO:tensorflow:batch_all_reduce: 158 all-reduces with algorithm = nccl, num_packs = 1
---------------------------------------------------------------------------
NotFoundError                             Traceback (most recent call last)
<ipython-input-5-ca3f8dc7254f> in <module>
----> 1 model.fit(dataset, epochs=num_epochs)

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)
   1083                 _r=1):
   1084               callbacks.on_train_batch_begin(step)
-> 1085               tmp_logs = self.train_function(iterator)
   1086               if data_handler.should_sync:
   1087                 context.async_wait()

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py in __call__(self, *args, **kwds)
    826     tracing_count = self.experimental_get_tracing_count()
    827     with trace.Trace(self._name) as tm:
--> 828       result = self._call(*args, **kwds)
    829       compiler = "xla" if self._experimental_compile else "nonXla"
    830       new_tracing_count = self.experimental_get_tracing_count()

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py in _call(self, *args, **kwds)
    886         # Lifting succeeded, so variables are initialized and we can run the
    887         # stateless function.
--> 888         return self._stateless_fn(*args, **kwds)
    889     else:
    890       _, _, _, filtered_flat_args = \

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/eager/function.py in __call__(self, *args, **kwargs)
   2948        filtered_flat_args) = self._maybe_define_function(args, kwargs)
   2949     return graph_function._call_flat(
-> 2950         filtered_flat_args, captured_inputs=graph_function.captured_inputs)  # pylint: disable=protected-access
   2951 
   2952   @property

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/eager/function.py in _call_flat(self, args, captured_inputs, cancellation_manager)
   1926       # No tape is watching; skip to running the function.
   1927       return self._build_call_outputs(self._inference_function.call(
-> 1928           ctx, args, cancellation_manager=cancellation_manager))
   1929     forward_backward = self._select_forward_and_backward_functions(
   1930         args,

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/eager/function.py in call(self, ctx, args, cancellation_manager)
    559               inputs=args,
    560               attrs=attrs,
--> 561               ctx=ctx)
    562         else:
    563           outputs = execute.execute_with_cancellation(

~/anaconda3/envs/sshtf24/lib/python3.7/site-packages/tensorflow/python/eager/execute.py in quick_execute(op_name, num_outputs, inputs, attrs, ctx, name)
     58     ctx.ensure_initialized()
     59     tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
---> 60                                         inputs, attrs, num_outputs)
     61   except core._NotOkStatusException as e:
     62     if name is not None:

NotFoundError: 3 root error(s) found.
  (0) Not found:  No algorithm worked!
	 [[node mobilenetv2_1.00_224/Conv1/Conv2D (defined at /home/librazxc/anaconda3/envs/sshtf24/lib/python3.7/threading.py:926) ]]
	 [[div_no_nan_1/ReadVariableOp_1/_264]]
  (1) Not found:  No algorithm worked!
	 [[node mobilenetv2_1.00_224/Conv1/Conv2D (defined at /home/librazxc/anaconda3/envs/sshtf24/lib/python3.7/threading.py:926) ]]
  (2) Not found:  No algorithm worked!
	 [[node mobilenetv2_1.00_224/Conv1/Conv2D (defined at /home/librazxc/anaconda3/envs/sshtf24/lib/python3.7/threading.py:926) ]]
	 [[Adam/Adam/group_deps/NoOp/_307]]
0 successful operations.
0 derived errors ignored. [Op:__inference_train_function_44264]

Errors may have originated from an input operation.
Input Source operations connected to node mobilenetv2_1.00_224/Conv1/Conv2D:
 cond_2/Identity (defined at <ipython-input-5-ca3f8dc7254f>:1)

Input Source operations connected to node mobilenetv2_1.00_224/Conv1/Conv2D:
 cond_2/Identity (defined at <ipython-input-5-ca3f8dc7254f>:1)

Input Source operations connected to node mobilenetv2_1.00_224/Conv1/Conv2D:
 cond_2/Identity (defined at <ipython-input-5-ca3f8dc7254f>:1)

Function call stack:
train_function -> train_function -> train_function
</code></pre>
</details>

Can you give me some suggestions on how to use multiple gpus in tf2.4?

**Comment by nikitamaia** [comment]
Hi @PureHing, I am unable to reproduce this error in Colab. Please [see the gist here](https://colab.research.google.com/gist/nikitamaia/3e602bdc300f4526e83ea03f0ac8c26a/untitled9.ipynb).
The Colab GPU runtime only has a single GPU, but I was also able to run this code without error with two P100s on GCP.

I wonder if this is an issue with your GPUs. I did a search for this error message and it seems it might have to do with incompatible versions of CUDA/cuDNN/TF. Please check and see if you are using a tested [build configuration.](https://www.tensorflow.org/install/source#linux)

**Comment by PureHing** [comment]
@nikitamaia Hi,Because I am using the RTX30 series of graphics cards, I cannot call gpu after installing the existing release version of tensorflow ([my issue](https://github.com/tensorflow/tensorflow/issues/43701#issuecomment-706089283)). Therefore, I installed the tf-nightly-gpu version, the above problem occurred when calling multiple gpus. I also tried to use it by building tensorflow 2.4.0 from source code, the same error occurs, and there is no problem in using a single gpu.
